cmd/tsdbrelay/denormalization.go

```diff
@@ -0,0 +1,69 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	"bosun.org/opentsdb"
+)
+
+type DenormalizationRule struct {
+	Metric   string
+	TagNames []string
+}
+
+func (d *DenormalizationRule) String() string {
+	inputTags, outputTags := "", ""
+	val := 'a'
+	for i, tagk := range d.TagNames {
+		if i != 0 {
+			inputTags += ","
+			outputTags += "."
+		}
+		inputTags += fmt.Sprintf("%s=%s", tagk, string(val))
+		outputTags += fmt.Sprintf("%s", string(val))
+		val++
+	}
+	return fmt.Sprintf("%s{%s} -> __%s.%s", d.Metric, inputTags, outputTags, d.Metric)
+}
+
+var denormalizationRules map[string]*DenormalizationRule
+
+func parseDenormalizationRules() {
+	if *denormalize == "" {
+		return
+	}
+	denormalizationRules = make(map[string]*DenormalizationRule)
+	rules := strings.Split(*denormalize, ",")
+	for _, r := range rules {
+		parts := strings.Split(r, "__")
+		if len(parts) < 2 {
+			log.Fatal("Denormalization rules must have at least one tag name specified.")
```

Comment:
yes, but let's try to contain log.Fatal to main(). this should probably take a string, and return the rules, so that it's testable.